### PR TITLE
Restart with multigrid and SST

### PR DESCRIPTION
**Pull request by demanosalvas**
FIx a bug which influenced the convergence behavior when using multigrid and SST by allowing the turbulent solution vector to propagate into the coarser grids (during multigrid) at each iteration. In addition this fix includes an update in the variable names for the parallel output.

### CODE REVIEW

**Comment by talbring** [comment]
Hi David,

can you make sure to use 2 spaces in instead of tabs for indentation (we have to be a little bit more strict on this one...). It also makes it easier to see the actual changes :)

Thanks!

File: SU2_CFD/src/integration_time.cpp [hunk 1]
```diff
@@ -72,15 +73,23 @@ void CMultiGridIntegration::MultiGrid_Iteration(CGeometry ***geometry,
...
-  
-  if ((restart && (Iteration == config[iZone]->GetnStartUpIter())) || startup_multigrid) {
-    for (iMGLevel = 0; iMGLevel < config[iZone]->GetnMGLevels(); iMGLevel++) {
-      SetRestricted_Solution(RunTime_EqSystem, solver_container[iZone][iMGLevel][SolContainer_Position],
-                             solver_container[iZone][iMGLevel+1][SolContainer_Position],
-                             geometry[iZone][iMGLevel], geometry[iZone][iMGLevel+1], config[iZone]);
-    }
-  }
-  
+	/*-- Since the restart takes care of this I dont think is required, but we should check after the new restart routines are added ---*/
```
Comment (economon) [inline SU2_CFD/src/integration_time.cpp]: Yep, let's check this once we put in the new restart stuff after your fix.

File: SU2_CFD/src/integration_time.cpp [hunk 2]
```diff
@@ -44,7 +44,8 @@ void CMultiGridIntegration::MultiGrid_Iteration(CGeometry ***geometry,
                                                 unsigned short RunTime_EqSystem,
                                                 unsigned long Iteration,
                                                 unsigned short iZone) {
-  unsigned short FinestMesh, iMGLevel;
+	
+	unsigned short FinestMesh, iMGLevel;
```
Comment (economon) [inline SU2_CFD/src/integration_time.cpp]: Please double check 2 spaces vs tabs, as Tim suggested. If you need any help setting this up in Xcode, just let me know (it's in preferences, and you can use ctrl+i to auto indent highlighted code). You can submit more commits straight to this branch after cleaning it up.

File: SU2_CFD/src/output_structure.cpp [hunk 3]
```diff
@@ -10394,30 +10394,50 @@ void COutput::LoadLocalData_Flow(CConfig *config, CGeometry *geometry, CSolver *
     
     nVar_Par += 3;
     Variable_Names.push_back("Temperature");
-    Variable_Names.push_back("Pressure_Coefficient");
+		if (config->GetOutput_FileFormat() == PARAVIEW){
```
Comment (economon) [inline SU2_CFD/src/output_structure.cpp]: Thanks for adding these. Once the new restarts are in, let's check that this works with ParaView.

File: SU2_CFD/src/integration_time.cpp [hunk 4]
```diff
@@ -806,17 +814,15 @@ void CSingleGridIntegration::SingleGrid_Iteration(CGeometry ***geometry, CSolver
...
-      if ((config[iZone]->GetMGCycle() == FULLMG_CYCLE) || config[iZone]->GetLowFidelitySim()) {
-        SetRestricted_Solution(RunTime_EqSystem, solver_container[iZone][iMesh][SolContainer_Position], solver_container[iZone][iMesh+1][SolContainer_Position], geometry[iZone][iMesh], geometry[iZone][iMesh+1], config[iZone]);
-      }
-      SetRestricted_EddyVisc(RunTime_EqSystem, solver_container[iZone][iMesh][SolContainer_Position], solver_container[iZone][iMesh+1][SolContainer_Position], geometry[iZone][iMesh], geometry[iZone][iMesh+1], config[iZone]);
-    }
-  }
+  /*--- If turbulence model, copy the turbulence bariables to the coarse levels ---*/
   
+	if (RunTime_EqSystem == RUNTIME_TURB_SYS) {
+		for (iMesh = FinestMesh; iMesh < config[iZone]->GetnMGLevels(); iMesh++) {
```
Comment (economon) [inline SU2_CFD/src/integration_time.cpp]: Here's the fix! Thanks!

File: SU2_CFD/src/solver_direct_mean.cpp [hunk 5]
```diff
@@ -4325,8 +4322,9 @@ void CEulerSolver::SetInitialCondition(CGeometry **geometry, CSolver ***solver_c
    all the multigrid levels, this is important with the dual time strategy ---*/
   
   if (restart && (ExtIter == 0)) {
-    
     Solution = new su2double[nVar];
+		solver_container[0][FLOW_SOL]->Set_MPI_Solution(geometry[0], config);
```
Comment (economon) [inline SU2_CFD/src/solver_direct_mean.cpp]: Let's leave this here for now, but I moved this chunk of code into the solver preprocessing part of the driver in the forthcoming changes with the restarts.

File: SU2_IDE/Xcode/SU2_CFD.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -68,6 +67,7 @@
 		05F1089C1978D2AE00F2F288 /* fluid_model_pvdw.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 05F108981978D2AE00F2F288 /* fluid_model_pvdw.cpp */; };
 		05F1089D1978D2AE00F2F288 /* fluid_model.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 05F108991978D2AE00F2F288 /* fluid_model.cpp */; };
 		05F108A41978D2F200F2F288 /* transport_model.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 05F108A31978D2F200F2F288 /* transport_model.cpp */; };
+		63322ABE1E56257200D5B23A /* integration_time.cpp in Sources */ = {isa = PBXBuildFile; fileRef = 05E6DBBF17EB62A100FA1F7E /* integration_time.cpp */; };
```
Comment (economon) [inline SU2_IDE/Xcode/SU2_CFD.xcodeproj/project.pbxproj]: Can you please revert the change here to the Xcode project if this was unintentional?

Comment (demanosalvas) [inline SU2_IDE/Xcode/SU2_CFD.xcodeproj/project.pbxproj]: This was definitely unintentional, and I think it just reflects the setting change from using tabs to spaces. I will try to revert it.

**Comment by demanosalvas** [comment]
Are there any other comments or suggestions?
